docs/design/autoware-architecture/index.md

```diff
@@ -0,0 +1,43 @@
+# Architecture overview
+
+This page describes the architecture of Autoware.
+
+## Introduction
+
+The current Autoware is defined to be a layered architecture that clarifies each module's role and simplifies the interface between them. By doing so:
+
+- Autoware's internal processing becomes more transparent.
+- Collaborative development is made easier because of the reduced interdependency between modules.
+- Users can easily replace an existing module (e.g. localization) with their own software component by simply wrapping their software to fit in with Autoware's interface.
+
+Note that the initial focus of this architecture design was solely on driving capability, and so the following features were left as future work:
+
+- Fail safe
+- Human Machine Interface
+- Real-time processing
+- Redundant system
+- State monitoring system
+
+## High-level architecture design
+
+![Overview](image/autoware-architecture-overview.drawio.svg)
+
+Autoware's architecture consists of the following six stacks. Each linked page contains a more detailed set of requirements and use cases specific to that stack:
+
+- [Sensing design](sensing/index.md)
+- [Map design](map/index.md)
+- [Localization design](localization/index.md)
+- [Perception design](perception/index.md)
+- [Planning design](planning/index.md)
+- [Control design](control/index.md)
+- [VehicleInterface design](vehicle/index.md)
```

Comment:
```suggestion
- [Vehicle Interface design](vehicle/index.md)
```

Reply:
@TakaHoribe Just one last fix and then I think this PR is good to go!

Reply:
Thank you! Done!

(... How great if I could push just a `commit suggestion` button.)

Reply:
@TakaHoribe didn't the button show up for you?

Reply:
The commit with suggestion button will cause CI failure.

Reply:
If it's DCO that fails then if you add your sign-off (even if it's someone else's suggestion) it should pass.

Reply:
Oh really? I didn't know that sign-off works for a suggestion. I'll try it next time

Reply:
Thank you!